### PR TITLE
TS-19-v2-26 Fix ESLint errors in containers directory

### PR DESCRIPTION
**Pull request by Niknikson**
Refactor was done in DepartmentPage, GroupList, LessonPage, SemesterPage, TeachersList.

`In rest of pages was fixed just eslint errors.`

### CODE REVIEW

File: frontend/src/containers/Auth/Auth.js [hunk 1]
```diff
@@ -131,19 +129,14 @@ const Auth = (props) => {
...
-        const { token } = props;
-        isSuccess = !!token;
+        isSuccess = !!props.token;
         message = t('successful_login_message');
         handleSnackbarOpenService(true, snackbarTypes.SUCCESS, message);
     }
 
     const commonCondition = !error && isSuccess && !isLoading;
-
+    // switch case
```
Comment (Akr1s) [inline frontend/src/containers/Auth/Auth.js:139]: Comment

File: frontend/src/containers/Auth/Auth.js [hunk 2]
```diff
@@ -50,32 +50,33 @@ const Auth = (props) => {
...
                         isToken = true;
-                        token = splitedParam[1];
+                        splitedParamToken = splitedParam;
                     }
                 }
             });
         }
-        if (social && isToken) socialLoginHandler({ authType: 'google', token });
+        if (social && isToken)
+            socialLoginHandler({ authType: 'google', token: splitedParamToken[1] });
```
Comment (Akr1s) [inline frontend/src/containers/Auth/Auth.js:72]: There is a constant for `google`

File: frontend/src/containers/Auth/Auth.js [hunk 3]
```diff
@@ -84,10 +85,7 @@ const Auth = (props) => {
     }, [props.response]);
 
     useEffect(() => {
-        if (
-            props.resetPasswordResponse &&
-            props.resetPasswordResponse.data.hasOwnProperty('message')
-        ) {
+        if (props.resetPasswordResponse && get(props.resetPasswordResponse.data, 'message')) {
```
Comment (Akr1s) [inline frontend/src/containers/Auth/Auth.js:88]: Destructure `props` in this file

File: frontend/src/containers/DepartmentPage/DepartmentPage.js [hunk 4]
```diff
@@ -19,135 +28,118 @@ import {
...
+    const [departmentCard, setDepartmentCard] = useState({ id: -1, disabledStatus: null });
     const [teacherDialog, setTeacherDialog] = useState(false);
-    const [editDepartment, setEditDepartment] = useState(false);
-    const { isSnackbarOpen, snackbarType, snackbarMessage, teachers } = props;
-    useEffect(() => clearDepartmentForm(), []);
+    const [isOpenConfirmDialog, setIsOpenConfirmDialog] = useState(false);
+
+    const clearDepartmentForm = () => {
+        clearDepartment();
+    };
```
Comment (Akr1s) [inline frontend/src/containers/DepartmentPage/DepartmentPage.js:51]: Unnecessary wrapper

File: frontend/src/containers/GroupList/GroupList.js [hunk 5]
```diff
@@ -24,188 +34,108 @@ import {
...
-        handleClickOpen(groupId);
+    const changeDisable = () => {
+        setIsDisabled((prev) => !prev);
     };
-    const getGroupTitle = (title) => {
-        const MAX_LENGTH = 5;
-        return getShortTitle(title, MAX_LENGTH);
+    const handleSnackbarClose = () => {
+        handleSnackbarCloseService();
     };
```
Comment (Akr1s) [inline frontend/src/containers/GroupList/GroupList.js:128]: Unnecessary wrapper

File: frontend/src/containers/LessonPage/LessonPage.js [hunk 6]
```diff
@@ -148,7 +109,7 @@ const LessonPage = (props) => {
 
     const defaultProps = {
         options: groups,
-        getOptionLabel: (option) => (option ? option.title : ''),
+        getOptionLabel: (option) => option && option.title,
```
Comment (Akr1s) [inline frontend/src/containers/LessonPage/LessonPage.js:112]: Label can be either `string` or `boolean`?

File: frontend/src/containers/SemesterPage/SemesterPage.js [hunk 7]
```diff
@@ -193,58 +160,80 @@ const SemesterPage = (props) => {
...
 
     const handleSemesterArchivedPreview = (semesterId) => {
         viewArchivedSemester(+semesterId);
     };
-    const setClassNameForDefaultSemester = (semester) => {
+    const setClassNameForDefaultSemester = (currentSemester) => {
         const defaultSemesterName = 'default';
         const className = 'svg-btn edit-btn';
-        return semester.defaultSemester === true
+        return currentSemester.defaultSemester === true
```
Comment (Akr1s) [inline frontend/src/containers/SemesterPage/SemesterPage.js:198]: Comparison to `true`

File: frontend/src/containers/SemesterPage/SemesterPage.js [hunk 8]
```diff
@@ -193,58 +160,80 @@ const SemesterPage = (props) => {
...
             ? `${className} ${defaultSemesterName}`
             : className;
     };
 
+    const handleSnackbarClose = () => {
+        handleSnackbarCloseService();
+    };
```
Comment (Akr1s) [inline frontend/src/containers/SemesterPage/SemesterPage.js:205]: Unnecessary wrapper

File: frontend/src/components/WeekRoomInfo/WeekRoomInfo.jsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+const WeekRoomInfo = ({ currentSemester, schedule, index, type = 'odd' }) => {
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/WeekRoomInfo/WeekRoomInfo.jsx:3]: need to refactor, reassign props 'type'

File: frontend/src/containers/Auth/Auth.js [hunk 10]
```diff
@@ -50,32 +50,33 @@ const Auth = (props) => {
...
+        if (social && isToken)
+            socialLoginHandler({ authType: 'google', token: splitedParamToken[1] });
     }
 
     useEffect(() => {
         if (
             authType === authTypes.REGISTRATION &&
             props.response &&
-            props.response.data.hasOwnProperty('message')
+            get(props.resetPasswordResponse.data, 'message')
```
Comment (ViktoriiaKosovan) [inline frontend/src/containers/Auth/Auth.js:79]: need to destructuring props

File: frontend/src/containers/Auth/Auth.js (hunk 3 above)
Comment (ViktoriiaKosovan) [inline frontend/src/containers/Auth/Auth.js:88]: as above

File: frontend/src/containers/Auth/Auth.js [hunk 11]
```diff
@@ -131,19 +129,14 @@ const Auth = (props) => {
         resetFormHandler(RESET_PASSWORD_FORM);
     };
 
-    let isSuccess;
-    let message;
-    const isLoading = props.loading;
-
     if (!error && props.userRole) {
```
Comment (ViktoriiaKosovan) [inline frontend/src/containers/Auth/Auth.js:132]: as above

File: frontend/src/containers/Auth/Auth.js [hunk 12]
```diff
@@ -131,19 +129,14 @@ const Auth = (props) => {
...
-        const { token } = props;
-        isSuccess = !!token;
+        isSuccess = !!props.token;
```
Comment (ViktoriiaKosovan) [inline frontend/src/containers/Auth/Auth.js:133]: need to destructuring props throughout all containers folder

**Pull request merged**